manager/api/requests.go

```diff
@@ -110,16 +112,20 @@ func (req viewResourceReq) validate() error {
 
 type listResourcesReq struct {
 	key    string
-	size   int
 	offset int
+	limit  int
 }
 
-func (req listResourcesReq) validate() error {
+func (req *listResourcesReq) validate() error {
 	if req.key == "" {
 		return manager.ErrUnauthorizedAccess
 	}
 
-	if req.size > 0 && req.offset >= 0 {
+	if req.limit > maxLimitSize {
```

Comment:
Reject the request if this is the case. Once again, **don't forget to update the API documentation**.